docs/grammar.md

```diff
@@ -297,12 +304,24 @@ Note that `%ignore` directives cannot be imported. Imported rules will abide by
 
 Declare a terminal without defining it. Useful for plugins.
 
+**Syntax:**
+```html
+%declare <TERMINAL>
+%declare <rule>
```

Comment:
`%declare <rule>` is not actually useable and should also be rejected by `lark.lark` and `load_grammar` (currently it causes an ugly `AttributeError: 'NoneType'`) I think changing the syntax is fine here, unless @erezsh knows of some way to actually use this.

Reply:
Since `%declare` was in both `load_grammar.py` and `lark.lark` before I started, how about we leave that for a new Issue?  I expect someone more familiar with `load_grammar.py` would be better at cutting `%declare rule` out of it than I will.

Reply:
@MegaIng I agree, I can't see a way in which declaring a rule is useful, or even makes sense.

Personally instead of removing it from the grammar, I would just add validation code.  @RossPatterson  I can write the validation code for load_grammar myself, if you prefer. Although it's not exactly rocket science :)

Reply:
> @MegaIng I agree, I can't see a way in which declaring a rule is useful, or even makes sense.
> 
> Personally instead of removing it from the grammar, I would just add validation code. @RossPatterson I can write the validation code for load_grammar myself, if you prefer. Although it's not exactly rocket science :)

It proved easier than I expected.  Fixed in 9bf7ddf466979c2f61d6e67d9c4b1e19b3f65e0d.